arrivals-api/src/main/scala/com/pagerduty/arrivals/api/proxy/Upstream.scala

```diff
@@ -12,18 +12,18 @@ import akka.http.scaladsl.model.Uri.Authority
 //    An HTTPRequest needs a place to go, you're job as the Upstream implemeter
 //    is to attach a uri to given request
 //
-// 2. prepare request.
+// 2. filter request.
 //    Last chance to make any modifications to the HTTPRequest before the request is made.
 //
-// 3. transform response.
-//    We've received a response from the prepared request, define a transform before we pass it
+// 3. filter response.
+//    We've received a response from the prepared request, define a filter before we pass it
 //    up to whomever made the request.
 trait Upstream[-AddressingConfig] {
   def metricsTag: String
 
   def addressRequest(request: HttpRequest, addressingConfig: AddressingConfig): HttpRequest
 
-  def prepareRequestForDelivery(request: HttpRequest): HttpRequest = request
+  def requestFilter: RequestFilter[Unit] = NoOpRequestFilter
```

Comment:
The verbs don't apply here, since calling the method does nothing but return the filter. It doesn't apply anything like previously.

In `HttpProxy`, there's `apply` implicitly called, it's `upstream.requestFilter.apply(addressedRequest, ()`.

Reply:
Ah ok, I did stumble across the signature here, but I didn't think of the implicit call to `apply` in the `HttpProxy`.